src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -318,6 +320,27 @@ private void Populate(ref LocalState state, int start)
             }
         }
 
+        // PROTOTYPE(NullableReferenceTypes): Consider setting treatUnconstrainedTypeParameterAsNullable during initial binding, and removing this method.
+        /// <summary>
+        /// Adjust the given type, treating unconstrained type parameters as nullable if in the appropriate context. This overrides the current state
```

Comment:
>unconstrained type parameters [](start = 44, length = 29)

This comment, and the method name, imply all unconstrained type parameters in `initialType` are adjusted, but we're only adjusting `initialType` if it is simply an unconstrained type parameter. Consider using singular here and in method name.